python/metatensor-torch/metatensor/torch/atomistic/outputs.py

```diff
@@ -75,7 +75,9 @@ def _check_energy(
         if selected_atoms is not None:
             expected_samples = expected_samples.intersection(selected_atoms)
 
-        if len(expected_samples.union(energy_block.samples)) != len(expected_samples):
+        if len(expected_samples.union(energy_block.samples.to(device="cpu"))) != len(
```

Comment:
instead of moving the data to CPU, I would rather create the `expected_samples` on the right device, replacing `torch.tensor(expected_values)` with `torch.tensor(expected_values, device=energy.device)`